libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/extensions/ContextEx.kt

```diff
@@ -20,7 +20,12 @@ fun Context.inferDeviceLanguage(): String = inferDeviceLocale().language
  */
 fun Context.inferDeviceLocale(): Locale =
     if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
-        this.resources.configuration.locales.get(0)
+        val locales = this.resources.configuration.locales
+        if (!locales.isEmpty && locales[0] != null) {
```

Comment:
Android 🤦 

```
/**
 * Get the locale list. This is the preferred way for getting the locales (instead of using
 * the direct accessor to {@link #locale}, which would only provide the primary locale).
 *
 * @return The locale list.
 */
public @NonNull LocaleList getLocales() {
    fixUpLocaleList();
    return mLocaleList;
}
```

They return a `@NonNull` `LocaleList` that can include `null` `Locale`s LMAO

Reply:
```
    /**
     * Returns true if the specified locale uses a 24-hour time format by default, ignoring user
     * settings.
     * @param locale the locale to check
     * @return true if the locale uses a 24 hour time format by default, false otherwise
     * @hide
     */
    public static boolean is24HourLocale(@NonNull Locale locale) {
        synchronized (sLocaleLock) {
            if (sIs24HourLocale != null && sIs24HourLocale.equals(locale)) {
                return sIs24Hour;
            }
        }

        final java.text.DateFormat natural =
                java.text.DateFormat.getTimeInstance(java.text.DateFormat.LONG, locale);
```

LYAO one more time 😆 

This method in the `DateFormat` class marks `locale` as **NonNull** and later reports NPE said that `locale == null` -> 👀  https://github.com/mapbox/mapbox-navigation-android/issues/2965#issuecomment-628533511